CHANGELOG.md

```diff
@@ -1,5 +1,101 @@
+* don't cleanup topics on engine close ([#4658](https://github.com/confluentinc/ksql/pull/4658)) ([#4857](https://github.com/confluentinc/ksql/issues/4857)) ([21e4751](https://github.com/confluentinc/ksql/commit/21e47519ffd8df8cfee083e559ac28f736f058c1))
+* use describeTopics() for Kafka healtcheck probe ([#4814](https://github.com/confluentinc/ksql/pull/4814)) ([578d0d5](https://github.com/confluentinc/ksql/commit/578d0d543bce685ca6e253a8c6f9bcfbf191093d))
+
+
+
+### chore
+
+* add GROUP BY support for any key names ([#4899](https://github.com/confluentinc/ksql/pull/4899)) ([e7cbdfc](https://github.com/confluentinc/ksql/commit/e7cbdfcc8c9853e2ae6dfcaf670e04f07ccd5444)), closes [#4898](https://github.com/confluentinc/ksql/issues/4898)
+* partition-by primitive key support (#4098) ([7addf88](https://github.com/confluentinc/ksql/commit/7addf8856a6d62a6890a5f2520eead26538233a6)), closes [#4098](https://github.com/confluentinc/ksql/issues/4098)
+
+
+
+### BREAKING CHANGES
+
+* Select star, i.e. `select *`, no longer expands to include `ROWTIME` column(s). Instead, `ROWTIME` is only included in the results of queries if explicitly included in the projection, e.g. `select rowtime, *`.
+This only affects new statements. Any view previously created via a `CREATE STREAM AS SELECT` or `CREATE TABLE AS SELECT` statement is unaffected.
+* This commit changes the system generated column name for any columns in projections that are struct field dereferences. Previously, the full path was used when generating the name, now only the final field name is used. For example, `SELECT someStruct->someField, ...` would previously of generated a column name of `SOMESTRUCT__SOMEFIELD`, and will now generate a name of `SOMEFIELD`. Generated column names may have a numeric appended to the end to ensure uniqueness, for example `SOMEFIELD_2`.
+Note: it is recommended that you do not rely on system generated column names for production systems as there naming may change between releases. Providing an explict alias ensures consistent naming across releases. For example, `SELECT someStruct->someField AS someField`.
+Backwards compatibility: existing running queries will not be affected by this change: they will continue to run with the same column names. Any statements executed after the upgrade will use the new names where no explicit alias is provided. Add explicit aliases to your statements if you require the old names, for example: `SELECT someStruct->someField AS SOMESTRUCT__SOMEFIELD, ...`
+* Existing queries that reference a single GROUP BY column in the projection would fail if they were resubmitted, due to a duplicate column. The same existing queries will continue to run if already running, i.e. this is only a change for newly submitted queries. Existing queries will use the old query semantics.
```

Comment:
Do this and the subsequent 3 bullets belong underneath the "This commit changes the..." bullet? If so they should be indented underneath it.

Reply:
This belongs to one commit. But the subsequent 3 bullets belong to another commit. I'll merged the 3 bullets below into one.

Reply:
This indentation still looks incorrect.

Reply:
I am not sure what indentation. Anyway, I fixed the one from the `Note: it is reco...` line. I hope that's the one I should've fixed.

Reply:
I see. These two bullets ("Existing queries that reference ..." and "Any existing persistent queries ... ") are confusing because they don't make sense by themselves, which is why Derek and I thought they were meant to be part of a previous entry, but it turns out they actually aren't -- they were just written in a way that refers to a previous part of the commit message which isn't included in the breaking change section, e.g., https://github.com/confluentinc/ksql/commit/2002458f6b668b7b17b05b192a7cc3fab0a23b20 (cc @big-andy-coates for future reference).

I think we should reword the second one to "Push queries may see changes in column names. Pull queries and any existing persistent queries are unaffected." to help clarify some of the confusion.